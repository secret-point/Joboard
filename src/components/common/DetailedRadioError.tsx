import React from "react";
import { Row } from "@amzn/stencil-react-components/layout";
import { CommonColors } from "../../utils/colors";
import { Status, StatusIndicator } from "@amzn/stencil-react-components/status-indicator";
import { translate as t } from "../../utils/translator";

interface DetailedRadioErrorProps {
  errorMessage?: string,
  errorMessageTranslationKey?: string
}

const DetailedRadioError = (props: DetailedRadioErrorProps) => {
  const { errorMessage, errorMessageTranslationKey } = props;

  const renderErrorMessage = () => {
    if(errorMessageTranslationKey && errorMessage) {
      return t(errorMessageTranslationKey, errorMessage);
    }

    return t("BB-Detailed-button-error-text-message", "Please check the box to proceed.");
  }

  return (
    <Row padding="S300" backgroundColor={CommonColors.RED05}>
      <StatusIndicator
        messageText={renderErrorMessage()}
        status={Status.Negative}
        iconAriaHidden={true}
      />
    </Row>
  )
}

export default DetailedRadioError;