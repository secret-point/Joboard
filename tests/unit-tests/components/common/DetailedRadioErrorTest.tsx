import React from "react";
import { shallow } from "enzyme";
import DetailedRadioError from "../../../../src/components/common/DetailedRadioError";

describe("DetailedRadioError", () => {
    it("should match snapshot", () => {
        const shallowWrapper = shallow(<DetailedRadioError/>);
    
        expect(shallowWrapper).toMatchSnapshot();
    });
});