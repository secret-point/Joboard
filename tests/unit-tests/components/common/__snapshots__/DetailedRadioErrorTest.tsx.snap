// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailedRadioError should match snapshot 1`] = `
<Row
  backgroundColor="#fef0ef"
  padding="S300"
>
  <StatusIndicator
    colorScheme="default"
    iconAriaHidden={true}
    messageText="Please check the box to proceed."
    size="small"
    status="negative"
  />
</Row>
`;
